From: Sameeksha Sankpal <sameekshasankpal@gmail.com>
Subject: init: mark envp_init as static
Date: Sat, 29 Nov 2025 00:46:50 +0530

Sparse reports a warning that 'envp_init' has no prior declaration:

  warning: symbol 'envp_init' was not declared. Should it be static?

The variable is only used within init/main.c and does not need external
linkage.  Mark it as static to limit its scope and fix the sparse warning.

Link: https://lkml.kernel.org/r/20251128191650.18821-1-sameekshasankpal@gmail.com
Signed-off-by: Sameeksha Sankpal <sameekshasankpal@gmail.com>
Cc: Alexander Shishkin <alexander.shishkin@linux.intel.com>
Cc: Christian Brauner <brauner@kernel.org>
Cc: Francesco Valla <francesco@valla.it>
Cc: guoweikang <guoweikang.kernel@gmail.com>
Cc: Huacai Chen <chenhuacai@kernel.org>
Cc: Huan Yang <link@vivo.com>
Cc: Ingo Molnar <mingo@kernel.org>
Cc: Lillian Berry <lillian@star-ark.net>
Cc: "Masami Hiramatsu (Google)" <mhiramat@kernel.org>
Cc: Mike Rapoport <rppt@kernel.org>
Cc: Thomas Gleinxer <tglx@linutronix.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 init/main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/init/main.c~init-mark-envp_init-as-static
+++ a/init/main.c
@@ -192,7 +192,7 @@ static int __init set_reset_devices(char
 __setup("reset_devices", set_reset_devices);
 
 static const char *argv_init[MAX_INIT_ARGS+2] = { "init", NULL, };
-const char *envp_init[MAX_INIT_ENVS+2] = { "HOME=/", "TERM=linux", NULL, };
+static const char *envp_init[MAX_INIT_ENVS+2] = { "HOME=/", "TERM=linux", NULL, };
 static const char *panic_later, *panic_param;
 
 static bool __init obsolete_checksetup(char *line)
_
